frodo.js

```diff
@@ -1,73 +1,74 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?